.github/workflows/main.yml

```diff
@@ -40,10 +45,33 @@ jobs:
            packaging
           python -m pip freeze
 
-      - name: Patch Package Versions
+      - name: Verify Changed files
+        uses: tj-actions/changed-files@v32
+        id: verify-changed-version
+        with:
+          files: |
+              odc/**
+              pyproject.toml
+              setup.py
+              setup.cfg
+
+      - name: Patch Package Versions when code change.
+        if: steps.verify-changed-version.outputs.any_changed == 'true'
+        id: patch-version
         run: |
-          find . -name _version.py | xargs python ./scripts/patch_version.py \
-           $(git rev-parse --short HEAD)
+          CTAG=$(git ls-remote --tags --sort='version:refname' | \
+            tail -1 | cut -d '/' -f3)
+          TAG=$(cat odc/stats/_version.py  | sed 's/[^0-9|.]//g')
+          if [[ $TAG == $CTAG || -z $CTAG ]]; then
+          # -1: auto increment of patch number
+          # other positive integers: set the patch number accordingly
+          python ./scripts/patch_version.py -1 odc/stats/_version.py
+          TAG=$(cat odc/stats/_version.py  | sed 's/[^0-9|.]//g')
+          git config --global user.email "$GITHUB_ACTOR@auto.noreply"
+          git config --global user.name "$GITHUB_ACTOR"
+          git commit -a -m "Bump version to $TAG"
+          git push origin HEAD:${{ github.head_ref }} --follow-tags
+          fi
```

Comment:
This segment is supposed to guarantee the version only got bumped up once no matter how many pushes or update on PR. It does multiple bumps because there isn't a tag yet. I'd be lazy on this, since the initial tag can be whatever as long as it matches `_version.py`.

Reply:
@omad the comment here. re
> I created a new branch based on this PR (#73), and was very surprised when I went to push to it a second time and found it had changed.
This Step had bumped the version number on me, inside a private branch, and will continue to bump the version number each time I push.